src/AutoSaveStatus/AutoSaveStatusProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { PropsWithChildren, useCallback, useEffect, useRef, useState } from "react";
+
+export enum AutoSaveStatus {
+  IDLE = "idle",
+  SAVING = "saving",
+  DONE = "done",
+  ERROR = "error",
+}
+
+export interface AutoSaveStatusContextType {
+  status: AutoSaveStatus;
+  /** Resets status to IDLE, particularly useful if "Error" or "Done" is stale */
+  resetStatus: VoidFunction;
+  errors: unknown[];
+  /** Notifies AutoSaveContext that a request is in-flight */
+  triggerAutoSave: VoidFunction;
+  /** Notifies AutoSaveContext that a request has settled, optionally taking an error */
+  resolveAutoSave: (error?: unknown) => void;
+}
+
+export const AutoSaveStatusContext = React.createContext<AutoSaveStatusContextType>({
+  status: AutoSaveStatus.IDLE,
+  resetStatus() {},
+  errors: [],
+  triggerAutoSave() {},
+  resolveAutoSave() {},
```

Comment:
Suggestion: Should we have these methods runtime fail with "btw you're missing an AutoSaveProvider?" Or I actually suppose we're okay if that happens, i.e. an embedded component down the tree can do an FYI about "hey I'm loading now" but :shrug: no one is watching it, that's fine.

Reply:
useFormState and useFormStates both call on this, and should be usable without a provider.

It's up to the person who wants the indicator (In this case, Beam itself) to wrap the Provider around the project. So it specifically (per a test) should not fail if the Provider isn't there.

Reply:
Yeah, makes sense; wouldn't mind an in-source comment calling out we purposefully want "it's fine to not have a provider"; several providers I've seen try to actively fail if used with just the default value, but just noting we purposefully don't want/need that here.